### PR TITLE
docs: update auth docs + add new storefront guides

### PR DESCRIPTION
**Pull request by shahednasser**
- Update auth docs with latest changes to authentication flows
- Add documentation for Google and GitHub providers
- Add storefront guides for social-login (using Google as an example)
- Update the OAS for auth routes.

> Note: This also includes documentation updates based on PR #9013. So, it should only be merged after that one is merged and published.

### CODE REVIEW

File: www/apps/resources/app/commerce-modules/auth/auth-flows/page.mdx [hunk 1]
```diff
@@ -77,46 +91,63 @@ Check out the [AuthIdentity](/references/auth/models/AuthIdentity) reference for
...
+
+// authentication successful
 ```
 
-It means the authentication process requires the user to perform an action with a third-party service. For example, when using the `google` provider, the user goes to the URL specified in the `location`'s value to log in with their Google account. 
+If the `authenticate` method returns a `location` property, the authentication process requires the user to perform an action with a third-party service. So, you return the `location` to the front-end or client to redirect to that URL.
+
+For example, when using the `google` provider, the `location` is the URL that the user is navigated to login.
+
+If the `authenticate` method doesn't return a `location` property and the returned `success` property, the user is successfully authenticated. This happens after the user logs-in with the provider, such as google, the first time.
```
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/auth-flows/page.mdx]: In the `authenticate` flow we always return a `location`, there is no way to have the user authenticated successfully at this point. That has to happen after calling the `/callback` route / sdk method

File: www/apps/resources/app/commerce-modules/auth/auth-flows/page.mdx [hunk 2]
```diff
@@ -77,46 +91,63 @@ Check out the [AuthIdentity](/references/auth/models/AuthIdentity) reference for
...
+    protocol,
+  }
+)
+
+if (success) {
+  // authentication succeeded
 }
 ```
 
+If the returned `success` property is `true`, the authentication with the third-party provider was successful. Subsequent authentication attempts can use the `authenticate` method only.
```
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/auth-flows/page.mdx]: Again, callback has to happen every time an oauth is triggered, `authenticate` is only there to construct the initial redirect URL

File: www/apps/resources/app/commerce-modules/auth/auth-providers/github/page.mdx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+<Note title="Tip">
+
+Learn about the authentication flow in [this guide](../../authentication-route/page.mdx).
+
+</Note>
+
+---
+
+## Install the Google Auth Module Provider
```
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/auth-providers/github/page.mdx]:
```suggestion
## Install the Github Auth Module Provider
```

File: www/apps/resources/app/commerce-modules/auth/auth-providers/github/page.mdx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+export const metadata = {
+  title: `GitHub Auth Module Provider`,
+}
+
+# {metadata.title}
+
+In this document, you’ll learn about the GitHub Auth Module Provider and how to install and use it in the Auth Module.
+
+The Google Auth Module Provider handles authenticating users with their GitHub account.
```
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/auth-providers/github/page.mdx]:
```suggestion
The Github Auth Module Provider handles authenticating users with their GitHub account.
```

There are maybe other places as well :)

File: www/apps/resources/app/commerce-modules/auth/authentication-route/page.mdx [hunk 5]
```diff
@@ -18,31 +18,68 @@ These routes are added by Medusa's application layer, not the Auth Module.
...
+4. The frontend sends a request to the [Callback Route](#callback-route) passing the `code` query parameter.
+5. If the callback validation is successful, the frontend receives the authentication token.
+6. Decode the received token in the frontend using tools like [react-jwt](https://www.npmjs.com/package/react-jwt).
+    - If the decoded data has an `actor_id` property, then the user is already registered. So, use this token for subsequent authenticated requests.
+    - If not, follow the rest of the steps.
+7. The frontend uses the authentication token to create the user with their respective API route.
+    - For example, for customers you would use the [Create Customer API route](!api!/store#customers_postcustomers).
+    - For admin users, you accept an invite using the [Accept Invite API route](!api!/admin#invites_postinvitesaccept)
+8. The frontend sends a request to the [Refresh Token Route](#refresh-token-route) to retrieve a new token for the user.
```
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/authentication-route/page.mdx]:
```suggestion
8. The frontend sends a request to the [Refresh Token Route](#refresh-token-route) to retrieve a new token with the user information populated.
```

File: www/apps/resources/app/commerce-modules/auth/authentication-route/page.mdx [hunk 6]
```diff
@@ -18,31 +18,68 @@ These routes are added by Medusa's application layer, not the Auth Module.
...
+
+After the first registration, you only use the [Auth Route](#auth-route) for subsequent authentication.
```
Comment (sradevski) [inline www/apps/resources/app/commerce-modules/auth/authentication-route/page.mdx]: Same here as above, we always use the callback

**Review by sradevski — COMMENTED** [review]
Hey Shahed let me know if you want to go through the oauth flow, just to make sure we are on the same page on how it works :)

File: www/apps/resources/app/commerce-modules/auth/auth-flows/page.mdx (hunk 1 above)
Comment (shahednasser) [reply www/apps/resources/app/commerce-modules/auth/auth-flows/page.mdx]: Ah I forgot to update this part! Will fix it

**Comment by shahednasser** [comment]
@sradevski thanks for the review! Actually I forgot to update some parts as I misunderstood the authentication flow at first (what I messaged you about basically 😄) so I forgot to update some parts that you caught. I think the rest of the PR conveys the correct flow, but please feel free to give it another review in case I missed something else

**Review by sradevski — APPROVED** [review]
LGTM, nice 👍

**Comment by shahednasser** [comment]
@olivermrbl we need to merge this one once the refresh token feature is out

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**